source/dofs/dof_tools_sparsity.cc

```diff
@@ -708,6 +667,84 @@ namespace DoFTools
 
 
 
+  template <int dim, int spacedim>
+  Table<2,bool>
+  bool_dof_couplings_from_component_couplings (const FiniteElement<dim,spacedim> &fe,
+                                               const Table<2,Coupling> &mask_in)
+  {
+    // build a mask for each dof, not like the one given which represents
+    // components. make sure we do the right thing also with respect to
+    // non-primitive shape functions, which takes some additional thought
+    Assert(mask_in.n_rows() == fe.n_components(),
+           ExcDimensionMismatch(mask_in.n_rows(),
+                                fe.n_components()));
+    Assert(mask_in.n_cols() == fe.n_components(),
+           ExcDimensionMismatch(mask_in.n_cols(),
+                                fe.n_components()));
+
+    const unsigned int n_dofs = fe.dofs_per_cell;
+
+    Table<2,bool> mask_out(n_dofs, n_dofs);
+
+    // check whether the table of couplings contains only true arguments,
+    // i.e., we do not exclude any index. that is the easy case, since we
+    // don't have to set up the tables
+    bool need_dof_mask = false;
+    for (unsigned int i=0; i<mask_in.n_rows(); ++i)
+      for (unsigned int j=0; j<mask_in.n_cols(); ++j)
+        if (mask_in(i,j) == none)
+          need_dof_mask = true;
```

Comment:
but why do you compare against `none`? can you elaborate in a comment when exactly you want to compute the DoF mask, given the different possibilities you might encounter in `mask_in`'s entries?